### DISCUSSION TITLE
DOMException: Failed to execute 'querySelector' on 'Document'

### DISCUSSION THREAD

**Discussion by saddomik** [ChromiumDotNet/WebView2.DevTools.Dom#12] [Q&A] [answered]
**Hi! I want to press this button:**

```c#
var btnTest = await ctx.QuerySelectorAsync<HtmlButtonElement>("#1461173234025-3129f8602eccbe259104553afa8415434b4581-02de_1461173234023-2568f8602eccbe259104553afa8415434b458-10ad");
```

![333222111](https://user-images.githubusercontent.com/62466045/218097025-e2a384d8-814d-435a-bf63-4586b0250c8f.png)

**I get an error:**

`System.Reflection.TargetInvocationException: "Exception has been thrown by the target of an invocation."
`
![553322](https://user-images.githubusercontent.com/62466045/218098286-b5392cbf-1df8-40b6-a717-a390e73bcdbe.png)

**What am I doing wrong? Help please**

[URL gointerract.io >](https://microsoft.gointeract.io/interact/index?interaction=1461173234028-3884f8602eccbe259104553afa8415434b4581-05d1&accountId=microsoft&loadFrom=CDN&appkey=196de13c-e946-4531-98f6-2719ec8405ce&Language=English&name=pana&CountryCode=en-US&Click%20To%20Call%20Caller%20Id=+16265860337&startedFromSmsToken=e9dmQhQ&dnis=1&token=UEuUbt)

**Comment by amaitland** [comment, marked as answer]
> What am I doing wrong?

As per the error message, your selector (`#1461173234025-3129f8602eccbe259104553afa8415434b4581-02de_1461173234023-2568f8602eccbe259104553afa8415434b458-10ad`) is not valid in the context of [document.querySelector](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector).

Run the following in the `DevTools Console` and you'll get the same exception.

```js
document.querySelector('#1461173234025-3129f8602eccbe259104553afa8415434b4581-02de_1461173234023-2568f8602eccbe259104553afa8415434b458-10ad');
```

Whilst the ID is valid, a CSS Id selector cannot start with a number. See https://stackoverflow.com/a/37271406/4583726 for more detail.

You'll need to use [document.getElementById](https://developer.mozilla.org/en-US/docs/Web/API/Document/getElementById) instead. There is no `API` method for that yet as querySelector is generally preferred, you can use something like the following though:

```c#
var btnTest = await ctx.EvaluateFunctionHandleAsync<HtmlButtonElement>("document.getElementById('1461173234025-3129f8602eccbe259104553afa8415434b4581-02de_1461173234023-2568f8602eccbe259104553afa8415434b458-10ad')");
```
